### DISCUSSION TITLE
Multiple Postgres errors from Rails application

### DISCUSSION THREAD

**Discussion by joshdvir** [postgresml/pgcat#710] [Q&A]
Hi, 

We have a rails application that does many queries to the database.
We installed Pgcat and it's mostly working great, but we see many errors in some of our Kafka consumers that are making many requests in a short period of time together to create a payload for Kafka.

The errors we get are:
1. `PG::ConnectionBad: PQsocket() can't get socket descriptor`
2. `PG::ConnectionBad: PQconsumeInput() server closed the connection unexpectedly  This probably means the server terminated abnormally before or while processing the request.`

I could not figure out where they were coming from.
In Rails, we have a pool of 20 connections, and we set idle_timeout to 5 minutes (default), where the pgcat timeout is set to 1 hour; I want Rails to disconnect the connection and not pgcat.

Our pgcat is in front of an RDS Aurora cluster with autoscaling enabled, and it scales up/down very well.

I got the pgcat logs in TRACE, and I don't see anything that would indicate any issues.

We love pgcat but having issues overcoming this one; any help would be appreciated.

**Comment by drdrsh** [comment]
These errors indicate that the connection was closed from either PgCat side or due to connection loss at TCP level (e.g. keepalive probe failure)

Do you have a network load balancer between your application and PgCat?  if so, please take a look at this article, you will likely find the information there relevant https://tech.instacart.com/the-vanishing-thread-and-postgresql-tcp-connection-parameters-93afc0e1208c

**Reply by drdrsh** [reply]
Do you see panics in the logs?
